PeripheryFunctions/TS_GetIDs.m

```diff
@@ -67,29 +73,27 @@
 case 'ops'
     theDataTable = Operations;
 otherwise
-    error('Specify ''ts'' or ''ops''');
+    error('Specify ''ts'', ''ops'', or ''opsName''');
 end
 
-% The cell of comma-delimited keyword strings:
-theKeywordCell = theDataTable.Keywords;
-
-% Split into sub-cells using comma delimiter:
-Keywords = SUB_cell2cellcell(theKeywordCell);
-
-% Find objects with a keyword that matchees that given:
-matches = cellfun(@(x)any(ismember(theKeyword,x)),Keywords);
+% OC: below has been changed so that we return a set in the same order as
+%       above, and nan's otherwise
+matches = nan(length(theFields),1);
```

Comment:
You say that `theFields` is a string in the input instructions, but here it seems like it's a cell?

Reply:
Ah yes, it is a cell array of strings (the Names to match). Starting to think we should definitely split this into two functions

Reply:
I'm open to that route…